src/main/java/codeu/model/data/Message.java

```diff
@@ -67,4 +67,24 @@ public String getContent() {
   public Instant getCreationTime() {
     return creation;
   }
+
+  /** Override equality, so messages are compared based on their ID */
+  @Override
+  public boolean equals(Object o) {
+
```

Comment:
Ditto here (for whitespace and redundant logic)

Reply:
Done.